src/Processors/Formats/RowInputFormatWithNamesAndTypes.h

```diff
@@ -62,24 +65,23 @@ class RowInputFormatWithNamesAndTypes : public RowInputFormatWithDiagnosticInfo
     virtual bool tryParseSuffixWithDiagnosticInfo(WriteBuffer &) { return true; }
     bool isGarbageAfterField(size_t, ReadBuffer::Position) override {return false; }
 
+
     /// Read row with names and return the list of them.
     virtual std::vector<String> readNames() = 0;
     /// Read row with types and return the list of them.
     virtual std::vector<String> readTypes() = 0;
 
+    virtual void addInputColumn(const String & column_name, std::vector<bool> & read_columns);
```

Comment:
No more need to make it virtual and protected

Reply:
done